### PR TITLE
dynamodb doc

### PR DESCRIPTION
**Pull request by razvan-moj-zz**

### CODE REVIEW

File: 02-deploying-an-app/008-add-dynamodb-to-your-app.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+1\. In order to create an DynamoDB cluster, you will need to work within our [cloud-platform-environments](https://github.com/ministryofjustice/cloud-platform-environments) repo. Git clone the repo and create a new branch.
+
+```bash
+
+  $ git clone git@github.com:ministryofjustice/cloud-platform-environments.git #git clone repo
+
+  $ cd cloud-platform-environments # navigate into cloud-platform-environments directory.
+
+  $ git co -b add_dynamodb   # create and checkout new branch.
```
Comment (jasonBirchall) [inline 02-deploying-an-app/008-add-dynamodb-to-your-app.md:16]: not a big issue but this is an alias.

File: 02-deploying-an-app/008-add-dynamodb-to-your-app.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+# Adding an DynamoDB cluster to your application
+
+## Introduction
+This document will cover how to add DynamoDB to your application using our [terraform module](https://github.com/ministryofjustice/cloud-platform-terraform-dynamodb-cluster).
+
+### Creating a cluster
+
+sddsf
```
Comment (jasonBirchall) [inline 02-deploying-an-app/008-add-dynamodb-to-your-app.md]: assuming this is a typo?

File: 02-deploying-an-app/008-add-dynamodb-to-your-app.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+1\. In order to create an DynamoDB cluster, you will need to work within our [cloud-platform-environments](https://github.com/ministryofjustice/cloud-platform-environments) repo. Git clone the repo and create a new branch.
+
+```bash
+
+  $ git clone git@github.com:ministryofjustice/cloud-platform-environments.git #git clone repo
+namespaces
+asa
```
Comment (jasonBirchall) [inline 02-deploying-an-app/008-add-dynamodb-to-your-app.md]: And here.

File: 02-deploying-an-app/008-add-dynamodb-to-your-app.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+3\. Create a terraform file within the resources directory.
+
+```bash
+
+  $ vi main.tf  #create a terraform file.
+
+```
+
+4\. Create your module and change the module name and input variables. More instructions [here](https://github.com/ministryofjustice/cloud-platform-terraform-dynamodb-cluster). Follow the example directory for more direction. Also create a Kubernetes secret, which will use the values from the Terraform output and add it to your namespace. More information [here](https://www.terraform.io/docs/providers/kubernetes/r/secret.html).
```
Comment (jasonBirchall) [inline 02-deploying-an-app/008-add-dynamodb-to-your-app.md:40]: More instructions here is pretty empty.

**Review by jasonBirchall — CHANGES_REQUESTED** [review]
Just a few changes. 
Also, is it worth putting the, 'how to get your db instance name' instructions in here?

**Review by jasonBirchall — APPROVED** [review]

File: 02-deploying-an-app/008-add-dynamodb-to-your-app.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+resource "kubernetes_secret" "dynamodb" {
+  metadata {
+    name      = "dynamodb"
+    namespace = "exampleapp"
+  }
+
+  data {
+    name = "${module.dynamodb.table_name}"
```
Comment (alkar) [inline 02-deploying-an-app/008-add-dynamodb-to-your-app.md:75]: `terraform fmt` please!

File: 02-deploying-an-app/008-add-dynamodb-to-your-app.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    access_key_id     = "${module.dynamodb.access_key_id}"
+    secret_access_key = "${module.dynamodb.secret_access_key}"
+  }
+}
+
+```
+
+The DB name will be present in the k8s secret, or if used outside the CP pipeline, create an output.tf file:
```
Comment (alkar) [inline 02-deploying-an-app/008-add-dynamodb-to-your-app.md:83]: The "out of pipeline" use can be confusing?
Do we really need it?

**Pull request merged**